modules/compliance-tests/src/smithy4s/compliancetests/internals/package.scala

```diff
@@ -29,4 +33,57 @@ package object internals {
       h.++(Hints(smithy.api.TimestampFormat.EPOCH_SECONDS.widen))
     )
   }
+
+  implicit class SchemaOps[A](val schema: Schema[A]) extends AnyVal {
+    def clearHints: Schema[A] =
+      schema.transformHintsTransitively(_ => Hints.empty)
+  }
+
+  def splitQuery(queryString: String): (String, String) = {
+    queryString.split("=", 2) match {
+      case Array(k, v) =>
+        (
+          k,
+          Uri.decode(
+            toDecode = v,
+            charset = StandardCharsets.UTF_8,
+            plusIsSpace = true
+          )
+        )
+      case Array(k) => (k, "")
+    }
+  }
+
+  def parseQueryParams(
+      queryParams: Option[List[String]]
+  ): ListMap[String, List[String]] = {
+    queryParams.combineAll
+      .map(splitQuery)
+      .foldLeft[ListMap[String, List[String]]](ListMap.empty) {
+        case (acc, (k, v)) =>
+          acc.get(k) match {
+            case Some(value) => acc + (k -> (value :+ v))
+            case None        => acc + (k -> List(v))
+          }
+      }
+  }
+
+  def extractHeaders(
+      maybeHeaders: Option[Map[String, String]]
+  ): Option[Headers] =
```

Comment:
I think you could have this one return an empty `Headers` instead of an `Option[Headers]`